src/AppVersion.hs

```diff
@@ -7,9 +7,19 @@ Get your application version
 -}
 
 module AppVersion
-       ( someFunc
-       ) where
+    ( getGitInfo
+    ) where
 
+import Control.Exception (throwIO)
 
-someFunc :: IO ()
-someFunc = putStrLn ("someFunc" :: String)
+import qualified GitHash as Git
+
+
+{- | Extract 'Git.GitInfo' about the current repository.
+-}
+getGitInfo :: IO Git.GitInfo
+getGitInfo = Git.getGitRoot "." >>= \case
+    Left err -> throwIO err
+    Right root -> Git.getGitInfo root >>= \case
+        Left err -> throwIO err
```

Comment:
Let's document what kind of errors can this `throw`.
Or what about providing another "safe" function? What do you think? What is usually done in such cases? It is the library, so I think that it makes sense to think more about handling these corner cases :thinking:

Reply:
@vrom911 Good idea. I guess, from the library point of view, it's better to provide a safer function that returns the error inside `Either`